### PR TITLE
Merge prio and prio-processor modules and use external Python bindings

### PR DESCRIPTION
**Pull request by acmiyaguchi**
This merges the `prio-processor` application module with the thin CLI wrapper found in `prio`, on the basis that the Python SWIG bindings are moved into libprio via https://github.com/mozilla/libprio/pull/98. This is done in preparation for the addition of a Spark module that implements the CLI interface.

### CODE REVIEW

**Comment by acmiyaguchi** [comment]
This is mostly ready for review. It looks like I need to look at how the google cloud SDK is installed. A large portion of the examples will break with this PR, although some of them are out of date or use the wrapper that I'd like to deprecate. What are your thoughts on removing the wrapper? It's the code under `prio_processor.prio.wrapper` at the moment, which is not being used by anything other than the examples.

**Comment by acmiyaguchi** [comment]
It's worth mentioning that the amount of black formatting ie very minimal. Additionally the base distribution here is CentOS, so the installation will be different (the SDK is already being installed, but it looks like the URL to the tarball is being rate limited?).

**Comment by acmiyaguchi** [comment]
I had to touch yet more files in order to make everything consistent with the library (with the exception of the browser-validation example which no longer can be run due to changing infrastructure).

File: Dockerfile [hunk 1]
```diff
@@ -5,56 +5,48 @@ ENV LANG en_US.utf8
...
 
-# Build the prio wrapper
-WORKDIR /app/prio
-RUN make
-
-WORKDIR /app/processor
-RUN python3 setup.py bdist_egg
-
-WORKDIR /app
+# NOTE: force reinstall is temporary until the package is published on pypi
```
Comment (wlach) [inline Dockerfile:41]: Is this still valid?

File: Dockerfile [hunk 2]
```diff
@@ -5,56 +5,48 @@ ENV LANG en_US.utf8
...
         pip3 install -r requirements.txt
 
 ENV SPARK_HOME=/usr/local/lib/python3.6/site-packages/pyspark
 ENV PYSPARK_PYTHON=python3
 
-CMD cd prio && tox && cd .. && \
-        cd processor && tox && cd .. \
-        prio/scripts/test-cli-integration && \
+CMD pytest -v tests && \
+        scripts/test-cli-integration && \
```
Comment (wlach) [inline Dockerfile:49]: Sorry if I've asked this before, but isn't it a little strange to run tests while building the docker container?

File: examples/asyncio/main.py [hunk 3]
```diff
@@ -78,6 +79,7 @@ async def client_produce(client, data_items, queue_a, queue_b, n_clients):
 
 
 async def main():
+    Prio_init()
```
Comment (wlach) [inline examples/asyncio/main.py:82]: kind of ashame we need to do this manually now (I assume there's a good reason)

File: examples/benchmarks/main.py [hunk 4]
```diff
@@ -3,10 +3,12 @@
 import matplotlib
 import matplotlib.pyplot as plt
 import numpy as np
-from prio import prio
+from prio_processor.prio import wrapper as prio
+from prio import PrioContext
 from tqdm import tqdm
 
 
+@PrioContext()
```
Comment (wlach) [inline examples/benchmarks/main.py:11]: Likewise, it seems ashame we need to use a decorator here but I trust your judgement if you say it's for the best.

File: requirements.in [hunk 5]
```diff
@@ -1,3 +1,4 @@
--e file:./prio
--e file:./processor
+-e file:.
 pytest
+pyyaml
+python-dotenv
```
Comment (wlach) [inline requirements.in:4]: Missing newline?

File: tests/resources/cli/server_a/intermediate/external/aggregate/data.ndjson [hunk 6]
```diff
@@ -0,0 +1 @@
+"AZWrRLClnH6BVNEhGPWrXlo8bE6FuRNoiiKrKF/0u6hYxk3pytSrKYEQ2Wp+wiZug0SrZFmiWnlc8I+nCAs="
```
Comment (wlach) [inline tests/resources/cli/server_a/intermediate/external/aggregate/data.ndjson:1]: Missing newline

File: Dockerfile (hunk 1 above)
Comment (acmiyaguchi) [reply Dockerfile:41]: No :)

File: Dockerfile (hunk 2 above)
Comment (acmiyaguchi) [reply Dockerfile:49]: This is `CMD` which sets the default command to be run on startup, which you might be confusing with `RUN`.

File: examples/asyncio/main.py (hunk 3 above)
Comment (acmiyaguchi) [reply examples/asyncio/main.py:82]: There is, it mostly deals with how NSS behaves under the hood. It's set up to be application global, and doesn't like it when the context is shutdown during the middle of execution. Hard to know when things get initialized during async execution (which is why I don't use the context manager here).

**Pull request merged**